lib/teiserver_web/live/account/profile/contributor.ex

```diff
@@ -73,15 +74,21 @@ defmodule TeiserverWeb.Account.ProfileLive.Contributor do
   end
 
   def handle_event("save-country_code", _, %{assigns: assigns} = socket) do
-    Account.update_user_stat(assigns.user.id, %{
-      "bar_plus.flag" => assigns.temp_country_code
-    })
-
-    Account.recache_user(assigns.user.id)
-
-    {:noreply,
-     socket
-     |> assign(:country_code, assigns.temp_country_code)}
+    if String.match?(assigns.temp_country_code, ~r/^\S+$/) do
```

Comment:
Too broad. Better something strict like `^[a-zA-Z0-9-]{2,8}$`, could be maybe even more strict then that.

Reply:
Updated regex to you suggestion. Not sure how to go more strict than that while allowing existing flags.